### PR TITLE
[node] fix types for glob exclude with file types

### PR DESCRIPTION
**Pull request by henriquekirchheck**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
    - https://github.com/nodejs/node/pull/52837
    - https://github.com/nodejs/node/blob/v22.9.0/lib/internal/fs/glob.js [Lines 350 and 545]
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

File: types/node/fs.d.ts [hunk 1]
```diff
@@ -4314,29 +4314,35 @@ declare module "fs" {
      */
     export function cpSync(source: string | URL, destination: string | URL, opts?: CopySyncOptions): void;
 
-    export interface GlobOptions {
+    interface GlobOptionsBase {
```
Comment (Renegade334) [inline types/node/fs.d.ts:4317]: Was there any problem with your original solution, having the derived interfaces inherit from `Omit<GlobOptions, "exclude">` and not needing an additional interface declaration?

**Comment by Renegade334** [comment]
Duplicate of #71165.

**Comment by henriquekirchheck** [comment]
> Duplicate of #71165.

I was unaware of the other pull request as I started this a few days ago and only committed after recently running the tests on my computer, I'll be closing this pr because I think the solution on https://github.com/DefinitelyTyped/DefinitelyTyped/pull/71165 is better.

Otherwise, I did add more tests for all possible combinations of calling glob, be it async, sync, or via callback, so maybe the tests there could be extended?

**Review by rbuckton — APPROVED** [review]

**Comment by henriquekirchheck** [comment]
Ready to merge

**Pull request merged**